src/components/blocked-patron/blocked-modal/BlockedModal.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC } from "react";
+import { getModalIds } from "../../../core/utils/helpers/general";
+import Modal from "../../../core/utils/modal";
+import { useText } from "../../../core/utils/text";
+import { useUrls } from "../../../core/utils/url";
+import { Link } from "../../atoms/link";
+
+interface BlockedModalProps {
+  blockedStatus: string;
+}
+
+const BlockedModal: FC<BlockedModalProps> = ({ blockedStatus }) => {
+  const t = useText();
+  const { blockedPatronELinkUrl } = useUrls();
+  const { blockedModal } = getModalIds();
+
+  return (
+    <Modal
+      modalId={blockedModal as string}
+      classNames="modal-loan"
+      closeModalAriaLabelText={t(`blockedPatronCloseModalAriaLabelText`)}
+      screenReaderModalDescriptionText={t(
+        `blockedPatronModalAriaDescriptionText`
+      )}
+    >
+      <div className="modal-text__container">
+        <h1 className="text-header-h3">
+          {t(`blockedPatron${blockedStatus}TitleText`)}
+        </h1>
+        <p className="mt-48 mb-48 text-body-large">
+          {t(`blockedPatron${blockedStatus}BodyText`)}
+        </p>
+        {blockedStatus === "E" && (
```

Comment:
I think we should make a clear type for this. String is not precise enough. And it is difficult for developers that does not know the domain to know what eg "E" is.
So either there should be a:
```typescript
export const PatronBlockType = {
  E: "PatronIsSomething"
  ..
} as const;
```
or:
```typescript
export enum PatronBlockType {
  E = 'PatronIsSomething',
  ...
}
````

Reply:
...or (I haven't looked it up) can we use something from the generated code/types ?

Reply:
added todo for now